website/static/js/contribAdder.js

```diff
@@ -373,17 +428,43 @@ var AddContributorViewModel = oop.extend(Paginator, {
             self.onInviteError.bind(self)
         );
     },
-    onInviteSuccess: function(result) {
+    onInviteSuccess: function (result) {
         var self = this;
         self.query('');
         self.results([]);
         self.page('whom');
         self.add(result.contributor);
     },
-    onInviteError: function(xhr) {
+    onInviteError: function (xhr) {
         var response = JSON.parse(xhr.responseText);
         // Update error message
         this.inviteError(response.message);
+    },
+    /**
+     * get node tree for treebeard from API V1
+     */
+    fetchNodeTree: function (treebeardUrl) {
+        var self = this;
+        return $.ajax({
+            url: treebeardUrl,
+            type: 'GET',
+            dataType: 'json'
+        }).done(function (response) {
+            self.nodesOriginal = projectSettingsTreebeardBase.getNodesOriginal(response[0], self.nodesOriginal);
+            self.hasChildren = (Object.keys(self.nodesOriginal).length > 1);
```

Comment:
This attribute doesn't seem to be initialized anywhere. That said, wouldn't it make this a method rather than an attribute?

```js
self.hasChildren = function() { Object.keys(self.nodesOriginal).length > 1); };
```